torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -664,6 +664,14 @@ class ShapePropagator {
         }
         return;
       }
+      case prim::grad: {
+        auto tt = node->input()->type()->cast<TensorType>();
```

Comment:
nit: you can remove the assert and just write `node->input()->type()->expect<TensorType>();`